pkg/emulator/libretro/manager/remotehttp/manager.go

```diff
@@ -0,0 +1,96 @@
+package remotehttp
+
+import (
+	"log"
+	"os"
+	"strings"
+
+	"github.com/giongto35/cloud-game/v2/pkg/config/emulator"
+	"github.com/giongto35/cloud-game/v2/pkg/downloader"
+	"github.com/giongto35/cloud-game/v2/pkg/emulator/libretro/core"
+	"github.com/giongto35/cloud-game/v2/pkg/emulator/libretro/manager"
+	"github.com/giongto35/cloud-game/v2/pkg/emulator/libretro/repo"
+	"github.com/giongto35/cloud-game/v2/pkg/emulator/libretro/repo/buildbot"
+	"github.com/giongto35/cloud-game/v2/pkg/emulator/libretro/repo/github"
+	"github.com/giongto35/cloud-game/v2/pkg/emulator/libretro/repo/raw"
+	"github.com/gofrs/flock"
+)
+
+type Manager struct {
+	manager.BasicManager
+
+	repo   repo.Repository
+	client downloader.Downloader
+	fmu    *flock.Flock
+}
+
+func NewRemoteHttpManager(conf emulator.LibretroConfig) Manager {
+	repoConf := conf.Cores.Repo
+
+	var repository repo.Repository
+	switch repoConf.Type {
+	case "raw":
+		repository = raw.NewRawRepo(repoConf.Url)
+	case "github":
+		repository = github.NewGithubRepo(repoConf.Url, repoConf.Compression)
+	case "buildbot":
+		fallthrough
+	default:
+		repository = buildbot.NewBuildbotRepo(repoConf.Url, repoConf.Compression)
+	}
+
+	// used for synchronization of multiple process
+	fileLock := os.TempDir() + string(os.PathSeparator) + "cloud_game.lock"
+
+	return Manager{
+		BasicManager: manager.BasicManager{
+			Conf: conf,
+		},
+		repo:   repository,
+		client: downloader.NewDefaultDownloader(),
+		fmu:    flock.New(fileLock),
+	}
+}
+
+func (m Manager) Sync() error {
+	declared := m.Conf.GetCores()
+	dir := m.Conf.GetCoresStorePath()
+
+	// IPC lock if multiple worker processes on the same machine
+	m.fmu.Lock()
```

Comment:
Cool!

Reply:
More cooler that it works at least in win and nix (: (didn't know that Windows has filelocks)